.github/workflows/build.yml

```diff
@@ -0,0 +1,222 @@
+name: CI Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+# cancel already running builds of the same branch or pull request
+concurrency:
+  group: ci-${{ github.head_ref }} || concat(${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CFLAGS: -g -O2 -Werror=pointer-arith -Werror=implicit-function-declaration
+  JOBS: 2
+  DEBUG: 1
+  LIBRARY_DEPS: |
+    libatk1.0-dev
+    libcairo2-dev
+    libdconf-dev
+    libgirepository1.0-dev
+    libglib2.0-dev
+    libgtk-3-dev
+    libgtk-layer-shell-dev
+    libice-dev
+    libmate-desktop-dev
+    libmate-menu-dev
+    libmateweather-dev
+    libpango1.0-dev
+    libsm-dev
+    libsoup2.4-dev
+    libwnck-3-dev
+    libx11-dev
+    libxrandr-dev
+    libgirepository1.0-dev
+  BUILD_DEPS: |
+    ccache
+    autoconf-archive
+    autopoint
+    gir1.2-freedesktop
+    git
+    gobject-introspection
+    gtk-doc-tools
+    lsb-release
+    make
+    mate-common
+    meson
+    yelp-tools
+    iso-codes
+    gobject-introspection
+  # TODO
+  SCAN_BUILD_DEPS: |
+    clang
+    clang-tools
+
+jobs:
+  build:
+    name: Build on Ubuntu
+    runs-on: ubuntu-latest
```

Comment:
ubuntu-latest is the tag of the github-runner as i understand docs, right? https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources.
Maybe using unbuntu:rolling as container for building? Same as you did for debian-testing.
In ubuntu:rolling we will see a new mayor mate release much quicker than in LTS, if they do updating LTS.

Reply:
I'm trying this, let's see if that works… if it does, it simplifies the build file :)

Reply:
Works and simplifies things, double win :)

Reply:
Cool, only one config :)